src/main/java/com/direwolf20/buildinggadgets/common/building/placement/ConnectedSurface.java

```diff
@@ -0,0 +1,145 @@
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.Queue;
+import java.util.function.Function;
+
+/**
+ * Surface that limits its attempt to blocks that is connected through either on its sides or corners. Candidates are
+ * selected from a wall region centered at a point and filtered with a 8-way adjacent flood fill.
+ *
+ * @see #iterator()
+ * @see Surface
+ */
+public final class ConnectedSurface implements IPlacementSequence {
+
+    /**
+     * @param world           Block access for searching reference
+     * @param searchingCenter Center of the searching region
+     * @param side            Facing to offset from the {@code searchingCenter} to get to the reference region center
+     */
+    public static ConnectedSurface create(IBlockAccess world, BlockPos searchingCenter, EnumFacing side, int range, boolean fuzzy) {
+        Region searchingRegion = Wall.clickedSide(searchingCenter, side, range).getBoundingBox();
+        return create(world, searchingRegion, pos -> pos.offset(side), searchingCenter, side, fuzzy);
+    }
+
+    public static ConnectedSurface create(IBlockAccess world, Region searchingRegion, Function<BlockPos, BlockPos> searching2referenceMapper, BlockPos searchingCenter, EnumFacing side, boolean fuzzy) {
+        return new ConnectedSurface(world, searchingRegion, searching2referenceMapper, searchingCenter, side, fuzzy);
+    }
+
+    private final IBlockAccess world;
+    private final Region searchingRegion;
+    private final Function<BlockPos, BlockPos> searching2referenceMapper;
+    private final BlockPos searchingCenter;
+    private final EnumFacing side;
+    private final boolean fuzzy;
+
+    @VisibleForTesting
+    private ConnectedSurface(IBlockAccess world, Region searchingRegion, Function<BlockPos, BlockPos> searching2referenceMapper, BlockPos searchingCenter, EnumFacing side, boolean fuzzy) {
+        this.world = world;
+        this.searchingRegion = searchingRegion;
+        this.searching2referenceMapper = searching2referenceMapper;
+        this.searchingCenter = searchingCenter;
+        this.side = side;
+        this.fuzzy = fuzzy;
+    }
+
+    /**
+     * The bounding box of the searchingRegion that is being searched.
+     */
+    @Override
+    public Region getBoundingBox() {
+        return searchingRegion;
+    }
+
+    /**
+     * {@inheritDoc}<br>
+     * <b>inaccurate representation (case 2)</b>:
+     *
+     * @return {@code getBoundingBox().contains(x, y, z)} since it would be costly to check each position
+     */
+    @Override
+    public boolean mayContain(int x, int y, int z) {
+        return searchingRegion.mayContain(x, y, z);
+    }
+
+    @Override
+    public IPlacementSequence copy() {
+        return new ConnectedSurface(world, searchingRegion, searching2referenceMapper, searchingCenter, side, fuzzy);
+    }
+
+    /**
+     * Uses a 8-way adjacent flood fill algorithm with Breath-First Search to identify blocks with a valid path. A position
```

Comment:
```suggestion
     * Uses a 8-way adjacent flood fill algorithm with Breadth-First Search to identify blocks with a valid path. A position
```
Now I understand what you were referring to with Breath-First... I really didn't get what breathing had to do with it, because of that typo XD